File: javascript/pangram/pangram.js
// @ts-check

/**
 * Checks if sentence is using each 1 of the 26 letters at least once.
 * @param {string} s the input text (case insensitive) to check
 * @returns {boolean} whether the given text is a pangram or not
 */
export const isPangram = s => new Set(s.toLowerCase().match(/[a-z]/g)).size==26;

File: fortran/pangram/pangram.f90
MODULE pangram; CONTAINS ! Whether given sentence `s` contains all 26 letters.
  PURE LOGICAL FUNCTION is_pangram(s); CHARACTER(*), INTENT(IN) :: s; mask = 0

    DO i = 1, len_trim(s); lo = ior(ichar(s(i:i)), ichar(' ')) ! lower (bit 5)
      IF (lo < iachar('a') .OR. lo > iachar('z')) CYCLE ! skip non-letter
      mask = ibset(mask, iachar('z') - lo); END DO ! set letter bit on `mask`

    is_pangram = mask == maskr(26); END FUNCTION; END MODULE ! all 26 letters
